elements/lisk-chain/src/diff.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+type HistoryType = [string?, number?];
+
+interface Frontier {
+	[key: string]: {
+		x: number;
+		history: HistoryType[];
+	};
+}
+
+const _diff = (initialBuffer: Buffer, finalBuffer: Buffer): HistoryType[] => {
+	const one = (idx: number): number => idx - 1;
+	const initial = Array.from(initialBuffer);
```

Comment:
do we need to convert this to array? same for below

Reply:
yeah,  we don't need it, earlier there were some operations related to arrays so I didn't change, but still I have to make below changes,

```typescript
if (initialBytesLength === 0) {
		return (Array.from(final)).map(b => ['+', b]);
	}

	if (finalBytesLength === 0) {
		return Array.from(initial).map(b => ['-', b]);
	}
```

Reply:
if `b` is number, you can just use map directly on Buffer. the iterator will return number (uint8)

Reply:
we want to return Array type and when I use the map on buffer it tries to modify the value of each element by something like `['+', 2]` and outputs a Buffer `Buffer <00, 00, 00, 00, 00>`